x/staking/types/authz.go

```diff
@@ -103,12 +103,10 @@ func (a StakeAuthorization) Accept(ctx sdk.Context, msg sdk.Msg) (authz.AcceptRe
 			Updated: &StakeAuthorization{Validators: a.GetValidators(), AuthorizationType: a.GetAuthorizationType()}}, nil
 	}
 
-	// check sufficient balance exists.
-	if _, isNegative := sdk.NewCoins(*a.MaxTokens).SafeSub(sdk.NewCoins(amount)); isNegative {
-		return authz.AcceptResponse{}, sdkerrors.ErrInsufficientFunds.Wrapf("amount is more than max tokens")
+	limitLeft, err := a.MaxTokens.SafeSub(amount)
+	if err != nil {
+		return authz.AcceptResponse{}, sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, "negative coin amount")
```

Comment:
there are two possible errors here with `SafeSub` can't use negative error description here I guess.
https://github.com/cosmos/cosmos-sdk/pull/11630/files#r850113361 if we do like this we can directly return the err here.

Reply:
Let's address this, and show the underlying error too, @NagaTulasi